src/kernel/src/fs/tmp/mod.rs

```diff
@@ -109,22 +107,36 @@ struct TempFs {
 
 impl Filesystem for TempFs {
     fn root(self: Arc<Self>, mnt: &Arc<Mount>) -> Result<Arc<Vnode>, Box<dyn Errno>> {
-        let vnode = alloc_vnode(mnt, &self, &self.root)?;
+        let vnode = self.alloc_vnode(mnt, &self.root)?;
 
         Ok(vnode)
     }
 }
 
-/// See tmpfs_alloc_vp
-#[allow(unused_variables)] // TODO: remove when implementing
-fn alloc_vnode(
-    mnt: &Arc<Mount>,
-    fs: &Arc<TempFs>,
-    node: &Arc<Node>,
-) -> Result<Arc<Vnode>, AllocVnodeError> {
-    // Make sure we don't return more than one Vnode for the same Node. It is a logic error if more
-    // than one vnode encapsulate the same node.
-    todo!()
+impl TempFs {
+    /// See tmpfs_alloc_vp on the PS4 for a reference.
+    fn alloc_vnode(
+        self: &Arc<TempFs>,
+        mnt: &Arc<Mount>,
+        node: &Arc<Node>,
+    ) -> Result<Arc<Vnode>, AllocVnodeError> {
+        let mut vnode_ref = node.vnode_mut();
+
+        if let Some(vnode) = vnode_ref.as_ref() {
+            Ok(vnode.clone())
+        } else {
+            let vnode = Vnode::new(
+                mnt,
+                node.ty().clone(),
+                "tmpfs",
+                VnodeBackend::new(node.clone()),
+            );
+
+            *vnode_ref = Some(vnode.clone());
+
+            Ok(vnode)
```

Comment:
What I saw in the PS4 is there are some logic before this.

Reply:
@SuchAFuriousDeath not sure if you are missed this one.

Reply:
If the missing parts here is not causing incorrect behavior we can put todo comment here.